### PR TITLE
Feat: mypy ci

### LINKED ISSUE #16: `mypy` CI
CI workflow to run `mypy .`, triggers on PR and Push events

### PR DESCRIPTION
**Pull request by luarss**
Fixes mypy CI for frontend and backend. 

Fixes #16

### CODE REVIEW

File: .github/workflows/mypy.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+on:
+  push:
+    branches:
+      - feat/mypy-ci # TODO: remove.
+      - master
+  pull_request:
+    branches:
+      - feat/mypy-ci # TODO: remove.
+      - master
```
Comment (vvbandeira) [inline .github/workflows/mypy.yaml:11]: Why not run for every branch?

File: .github/workflows/mypy.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          python-version: 3.12
+  
+      - name: Install dependencies
+        run: |
+          pip install -r frontend/requirements-test.txt
+      
+      - name: Run MyPy
+        run: |
+          python -m mypy --strict frontend
+          # python -m mypy --strict evaluation # TODO.
```
Comment (vvbandeira) [inline .github/workflows/mypy.yaml:51]: remove

File: .github/workflows/mypy.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  backend:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
```
Comment (vvbandeira) [inline .github/workflows/mypy.yaml:22]: Why are these in `v2` while #12 uses `v4`?

File: frontend/requirements-test.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+pytz==2024.1
+google-auth==2.30.0
+google-auth-httplib2==0.2.0
+google-auth-oauthlib==1.2.0
+gspread==6.1.2
+python-dotenv==1.0.1
+mypy
+flask
+types-pytz
+types-requests
```
Comment (luarss) [inline frontend/requirements-test.txt:14]: Pin this requirements

File: frontend/requirements.txt [hunk 5]
```diff
@@ -7,4 +7,5 @@ google-auth==2.30.0
 google-auth-httplib2==0.2.0
 google-auth-oauthlib==1.2.0
 gspread==6.1.2
-python-dotenv==1.0.1
\ No newline at end of file
+python-dotenv==1.0.1
+flask
```
Comment (luarss) [inline frontend/requirements.txt:11]: Pin this

File: .github/workflows/mypy.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        run: |
+          pip install -r backend/requirements-test.txt
+      - name: Run MyPy
+        run: |
+          python -m mypy --strict backend
+  frontend:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (luarss) [inline .github/workflows/mypy.yaml:25]: v4

Comment (vvbandeira) [inline .github/workflows/mypy.yaml:25]:
```suggestion
        uses: actions/checkout@v4
```

**Review by vvbandeira — DISMISSED** [review]

**Comment by vvbandeira** [comment]
@luarss 
This has conflicts due to the other PR merge.

**Review by vvbandeira — DISMISSED** [review]

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**